src/utils.rs

```diff
@@ -162,37 +162,37 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
 /// [`Transcript`].
 pub(crate) fn plusminus_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
 ) -> BigNumber {
     let mut is_neg_byte = vec![0u8; 1];
     transcript.challenge_bytes(b"sampling negation bit", is_neg_byte.as_mut_slice());
     let is_neg: bool = is_neg_byte[0] & 1 == 1;
 
     // The sampling method samples from the open interval, so add 1 to sample from
     // the _closed_ interval we want here.
-    let open_interval_max = n + 1;
+    let q = k256_order();
+    let open_interval_max = &q + 1;
     let b = positive_bn_random_from_transcript(transcript, &open_interval_max);
     match is_neg {
         true => -b,
         false => b,
     }
 }
 
-/// Derive a deterministic pseduorandom value in `[0, n)` from the
-/// [`Transcript`].
 pub(crate) fn positive_bn_random_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
+    _n: &BigNumber,
 ) -> BigNumber {
-    let len = n.to_bytes().len();
+    let len = _n.to_bytes().len();
     let mut t = vec![0u8; len];
     // To avoid sample bias, we can't take `t mod n`, because that would bias
     // smaller numbers. Instead, we re-sample a new value (different because
     // there's a new label in the transcript).
     loop {
```

Comment:
Here's something I didn't notice when I wrote the issue, but that we should fix while you're in here: this loops forever if something is wrong with the randomness, as you found. In other places we set a maximum # of loop iterations and throw an error if we exceed it. See `random_bn_in_z_star` as an example (it uses an iterator instead of a loop).

The steps to fix this would be
1. Update this function to return a `Result<BigNumber>`
1. Add a condition to the loop so it goes for a maximum of `CRYPTOGRAPHIC_RETRY_MAX` times
3. Return an `InternalError::RetryFailed` error if we exceed the maximum.
4. Update all the call sites to propagate the error upward with `?`

Reply:
It shouldn't have an infinite loop; the current version will almost always work unless something goes wrong with the randomness, but if the RNG is messed up it might fail forever, which is a bad user experience.

I don't have a strong preference between a bounded loop (like, `for _ in 0..RETRY_MAX { ... }`) vs the iterator approach below.